src/table.py

```diff
@@ -1489,44 +1525,57 @@ def top_row_is_bold(bbox):
         # return this if we determine that the top row is the header
         header_top_row = TableHeader(bbox, cells, self.extract()[0], False)
 
-        # one-line tables have no extra header
+        # 1-line tables have no extra header
         if len(self.rows) < 2:
             return header_top_row
 
-        # x-ccordinates of columns between x0 and x1 of the table
+        # 1-column tables have no extra header
         if len(cells) < 2:
             return header_top_row
 
-        col_x = [
-            c[2] if c is not None else None for c in cells[:-1]
-        ]  # column (x) coordinates
+        # assume top row is the header if second row is empty
+        row2 = self.rows[1]  # second row
+        if all([c is None for c in row2.cells]):  # pylint: disable=R1729
```

Comment:
Similar to above, suggest generator expression.

Reply:
I will use `if set(row2.cells) == {None}:` which does what is needed here.

Reply:
Really? That seems both obscure and inefficient.

Reply:
Hmm, that seems a bit obscure, but i'm ok with it if you insist.

Reply:
Interesting, the opposite of what i expected!

Reply:
Admittedly, this only plays any role for very long sequences. If however the list is really long and there is a high chance for non-equality in the first items, the generator reacts with shortcut exit and thus is preferrable.
All this under the assumption, that the list / sequence in question is not being created as part of the generator expression.

Just tried with setting item 100 to something else (not `None`). Then the generator was slightly faster than the set building.

Reply:
I think there are a lot of pros and cons. Let me make another change and use generators wherever possible. This will avoid doubts and discussions about the best way each time again.

Forgot to mention another advantage of generators: they do not depend on sequence items being hashable (in contrast to sets).